ref_builder/otu/utils.py

```diff
@@ -64,6 +64,26 @@ def check_isolate_size(
     )
 
 
+def get_segments_min_length(segments: list[Segment]) -> int:
+    """Return the shortest minimum length from a list of segments."""
+    shortest_segment = segments[0]
+    for segment in segments:
+        if segment.length < shortest_segment.length:
+            shortest_segment = segment
+
+    return shortest_segment.min_length
```

Comment:
```suggestion
return min(segments, key=lambda s: s.length).min_length
```
Do the same with `max`.

If these functions are only used in one place, I'd say they can be dropped and the one liner used in-place.